### PR TITLE
Misc. fixes for Coverity defects

### PR DESCRIPTION
**Pull request by peter-b**
Coverity-Id: 120052
Coverity-Id: 26003
Coverity-Id: 16051

### CODE REVIEW

File: engine/src/srvcgi.cpp [hunk 1]
```diff
@@ -1816,6 +1816,9 @@ static bool cgi_send_headers(void)
 			case kMCSOutputTextEncodingUTF8:
 				sprintf(t_content_header, "Content-Type: text/html; charset=utf-8\n");
 				break;
+			default:
```
Comment (montegoulding) [inline engine/src/srvcgi.cpp]: @peter-b is there a reason not to handle `kMCSOutputTextEncodingUndefined` here so when we turn on  `-Werror=switch` we we will get an error when we add an output encoding and don't handle it? I did handle this case in my switch patch but I didn't actually output a header as you are which seems sensible... or should this actually be an unreachable state?

Comment (montegoulding) [reply engine/src/srvcgi.cpp]: Actually I wonder if we can just remove `kMCSOutputTextEncodingUndefined` entirely... it doesn't appear there's a code pat that can give `MCserveroutputtextencoding` that value...

Comment (peter-b) [reply engine/src/srvcgi.cpp]: Sure, I'll look at both options. Good spot. 😎

**Comment by montegoulding** [comment]
restarting travis after an IDE test crash

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 5d59fd3

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 5d59fd352dd88cf5d9f65e4736a5871537761ae2

File: engine/src/osspec.h [hunk 2]
```diff
@@ -240,23 +240,21 @@ MCSErrorMode MCS_get_errormode(void);
...
 	kMCSOutputTextEncodingWindows1252,
 	kMCSOutputTextEncodingMacRoman,
 	kMCSOutputTextEncodingISO8859_1,
 	kMCSOutputTextEncodingUTF8,
-	
+};
+
 #if defined(__MACROMAN__)
-	kMCSOutputTextEncodingNative = kMCSOutputTextEncodingMacRoman,
+static const MCSOutputTextEncoding kMCSOutputTextEncodingNative = kMCSOutputTextEncodingMacRoman
```
Comment (livecodeali) [inline engine/src/osspec.h]: There's a missing semicolon here.

**Comment by livecode-vulcan** [comment]
😞 test failure 5d59fd352dd88cf5d9f65e4736a5871537761ae2
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/489)
- try-community-mac: [failure](https://vulcan.livecode.com/builders/try-community-mac/builds/494)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/489)
- try-community-emscripten: [pending](https://vulcan.livecode.com/builders/try-community-emscripten/builds/489)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/495)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/489)
- try-community-android-armv6: [pending](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/489)

**Comment by peter-b** [comment]
@livecodeali Thanks -- I've fixed that!

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay fa2e223

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok fa2e22387537898b611d4dfcef121c722b582f0c

**Comment by livecode-vulcan** [comment]
😎 test success fa2e22387537898b611d4dfcef121c722b582f0c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/491)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/496)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/491)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/491)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/497)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/491)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/491)

**Comment by peter-b** [comment]
@livecode-vulcan retry fa2e223

**Comment by livecode-vulcan** [comment]
😎 test success fa2e22387537898b611d4dfcef121c722b582f0c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/492)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/497)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/492)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/492)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/498)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/492)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/492)

**Pull request merged**